benchmark/solkz/compositional_field/CMakeLists.txt

```diff
@@ -33,7 +33,16 @@ ENDIF ()
 DEAL_II_INITIALIZE_CACHED_VARIABLES()
 
 SET(TARGET "solkz_compositional_field")
+SET(TARGET_DEPENDENCY "solkz")
+
 PROJECT(${TARGET})
 
+INCLUDE_DIRECTORIES("..")
+
 ADD_LIBRARY(${TARGET} SHARED solkz_compositional_field.cc)
-ASPECT_SETUP_PLUGIN(${TARGET})	
+ADD_LIBRARY(${TARGET_DEPENDENCY} SHARED ../solkz.cc)
```

Comment:
As you suggested, I attempted to import the header file via
`#include ASPECT_SOURCE_DIR "/benchmarks/solkz/solkz.cc"` 
however the gnu compiler does not concatenate the two strings properly even after resolving `ASPECT_SOURCE_DIR` correctly. I further investigated other possibilities including "stringification" as well as "concatenation" and kept running into the issue of tokenization of strings. For example, during preprocessing, certain symbols such as '.' are considered as operators and therefore "solkz.h" would be broken into three tokens. This is a problem when concatenating since the operation expects two characters. Ofcourse, I then looked into delimiting the operators such as '.', but I couldn't find any useful hints online.

I am wondering if a solution exists within CMake. For example, if `ASPECT_SOURCE_DIR` can be propagated as a cmake property during the call to `ASPECT_SETUP_PLUGIN(${TARGET})`. I am not sure if this is feasible however.